app/domain/authentication/authn_jwt/fetch_mandatory_claims_to_validate.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module Authentication
+  module AuthnJwt
+    OPTIONAL_CLAIMS = [ISS_CLAIM_NAME, EXP_CLAIM_NAME, NBF_CLAIM_NAME, IAT_CLAIM_NAME].freeze
+
+    # FetchMandatoryClaimsToValidate command class is responsible to return a list of mandatory JWT standard claims to
+    # validate.
+    FetchMandatoryClaimsToValidate ||= CommandClass.new(
+      dependencies: {
+        fetch_issuer_value: ::Authentication::AuthnJwt::FetchIssuerValue.new,
+        logger: Rails.logger
+      },
+      inputs: %i[authentication_parameters]
+    ) do
+
+      def call
+        @logger.debug(LogMessages::Authentication::AuthnJwt::FetchingJwtMandatoryClaimsToValidate.new)
+        fetch_mandatory_claims_to_validate
+        @logger.debug(LogMessages::Authentication::AuthnJwt::FetchedJwtMandatoryClaimsToValidate.new)
+
+        @mandatory_claims_to_validate_list
+      end
+
+      private
+
+      # fetch_mandatory_claims_to_validate function is responsible to return a list of mandatory JWT standard claims to
+      # validate, according to the following logic:
+      # For each optional claims (iss , exp,nbf, iat) which exists in the token, add to mandatory list
+      # Note: mandatory list also contains the value to validate if necessary (for example iss: cyberark.com)
+      def fetch_mandatory_claims_to_validate
+        @mandatory_claims_to_validate_list ||= []
+
+        OPTIONAL_CLAIMS.each do |optional_claim|
+          @logger.debug(LogMessages::Authentication::AuthnJwt::ValidatingJwtOptionalClaimToValidate.new(optional_claim))
+
+          if @authentication_parameters.decoded_token[optional_claim]
+            add_to_mandatory_claims_list(optional_claim)
+          end
+        end
+      end
+
+      def add_to_mandatory_claims_list(claim)
+        @logger.debug(LogMessages::Authentication::AuthnJwt::AddingJwtMandatoryClaimToValidate.new(claim))
+
+        @mandatory_claims_to_validate_list.push(::Authentication::AuthnJwt::JwtMandatoryClaim.new(name: claim, value: claim_value_to_validate(claim)))
+      end
+
+      def claim_value_to_validate(claim)
+        if claim == ISS_CLAIM_NAME
+          return @fetch_issuer_value.(@authentication_parameters)
+        end
+
+        nil
```

Comment:
what's this? are we planning to have other claims in the future? if so i would change the above to `switch-case` and put `nil` in the `else` (with a comment saying what's going on)